src/planscape/forsys/tests/test_evaluate_scenario.py

```diff
@@ -0,0 +1,166 @@
+
+    def test_evaluate_scenario_fake_run(self):
+        output = self._call_command(self.scenario.pk, '--fake_run', '--fake_run_time=1')
+        scenario_result = ScenarioResult.objects.get(scenario__id=self.scenario.pk)
+        self.assertEquals(scenario_result.status, ScenarioResultStatus.SUCCESS)
+        details = json.loads(scenario_result.run_details)
+        self.assertRegex(details['comment'], r'Fire Swamp')
+        self.assertTrue('start_time' in details)
+        self.assertTrue('end_time' in details)
+        self.assertTrue(details['elapsed_time'] > 1.0)
+
+    def test_evaluate_scenario_fake_run_with_failure(self):
+        output = self._call_command(self.scenario.pk, '--fake_run', '--fake_run_time=2', '--fake_run_failure_rate=100')
+        scenario_result = ScenarioResult.objects.get(scenario__id=self.scenario.pk)
+        self.assertEquals(scenario_result.status, ScenarioResultStatus.FAILURE)
+        details = json.loads(scenario_result.run_details)
+        self.assertRegex(details['comment'], r'land war in Asia')
+        self.assertTrue('start_time' in details)
+        self.assertTrue('end_time' in details)
+        self.assertTrue(details['elapsed_time'] > 1.0)
+
+    def test_missing_scenario_id(self):
+        try:
+            output = self._call_command()
+            raise ShouldHaveFailedException(output)
+        except ShouldHaveFailedException as e:
+            raise ShouldHaveFailedException(e)
+        except Exception as e:
+            self.assertRegex(str(e), r'the following arguments are required: scenario_id')
+
+    def test_nonexistent_scenario_id(self):
+        try:
+            output = self._call_command(999999)
+            raise ShouldHaveFailedException(output)
+        except ShouldHaveFailedException as e:
+            raise ShouldHaveFailedException(e)
+        except Exception as e:
+            self.assertRegex(str(e), r'ScenarioResult matching query does not exist')
+
+    def test_garbage_scenario_id(self):
+        try:
+            output = self._call_command('abcdef')
+            raise ShouldHaveFailedException(output)
+        except ShouldHaveFailedException as e:
+            raise ShouldHaveFailedException(e)
+        except Exception as e:
+            self.assertRegex(str(e), r'argument scenario_id: invalid int value')
+
+    def test_evaluate_scenario_ineligible_scenario_running(self):
+        scenario_result = ScenarioResult.objects.get(scenario__id=self.scenario.pk)
+        scenario_result.status = ScenarioResultStatus.RUNNING
+        scenario_result.save()
+        try:
+            output = self._call_command(self.scenario.pk, '--fake_run', '--fake_run_time=1')
+            raise ShouldHaveFailedException(output)
+        except ShouldHaveFailedException as e:
+            raise ShouldHaveFailedException(e)
+        except Exception as e:
+            self.assertRegex(str(e), r'ScenarioResult matching query does not exist')
```

Comment:
Would this exception message match the one that gets raised?  The Scenario Result does exist, it's just not pending (same q for the tests below)

Reply:
Since the code is actually filtering at the django-postgres request level, indeed, all we get is that no scenarioresult matched one (with the pending status).

Reply:
Thanks for the clarification! Then my only note is a future one that we'll probably want more specific exception messages (which I'm sure is already on ur radar)

Reply:
I changed the return exception payload to indicate that indeed a scenario has to be pending.  However, the DB lookup won't distinguish between that state and a nonexistent scenario ID, and so the pending message could still be superfluous.